Hazel/src/Hazel/Utility/Random.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include <random>
+
+namespace Hazel {
+
+	class Random
+	{
+	public:
+		/*
+		* Returns a Random Real between min and max
+		*/
+		template<typename T>
+		static T Real(T min, T max)
+		{
+			HZ_CORE_ASSERT(min < max, "min is >= max");
+			std::uniform_real_distribution<T> dist(min, max);
+			auto& mt = GetEngine();
+			return dist(mt);
+		}
+
+		/*
+		* Returns a Random Integer between min and max
+		*/
+		template<typename T>
+		static T Int(T min, T max)
```

Comment:
Maybe an alternative + integer type only and using the previous method (but again I'm not specialist on random number and template type). I made it to include `high` in random values (instead of excluded).
```
template<typename T>
typename std::enable_if<std::is_integral<T>::value && !std::is_floating_point<T>::value, T>::type
static T Int(T const low = zero<T>(), T const high = one<T>())
{
      return low + static_cast<T>(std::floor(Random::Real<float>() * static_cast<float>(high - low + 1)));
}
```

Reply:
if you are going to use it , 
i believe the second check for  **!std::is_floating_point** is unneccessary **std::integral** already covers it [](https://en.cppreference.com/w/cpp/types/is_integral)  

also you can shorten it by by using **std::is_integral_v** ; no need for the extra::type then  and same for enable_if_t . And i believe you do not need the second T if you are already providing the return type and if you use you dont have to specifiy the return type;
one more suggestion maybe move the whole check to template parameters;
```
template<typename T, typename =std::enable_if_t<std::is_integral_v<T>, T>> // the last T is for the return type 
static Int(....
```
or
```
template<typename T, typename =std::enable_if_t<std::is_integral_v<T>>> 
static  T Int(....
```
But might need to check as . Just a humble suggestion if used; those enables can be shortened by using an Alias we can define in the Core like;
**in Core.h define Alias;**
```
template<typename  T>
using EnableIfReal = std::enable_if_t<std::is_integral_v<T>>
```

**and actual code becomes shorter easy on the eyes;**
```
template<typename T, typename =EnableIfReal<T>> 
static  T Int(.....
```